Libraries/LibWeb/DOM/CharacterData.cpp

```diff
@@ -88,11 +88,15 @@ WebIDL::ExceptionOr<void> CharacterData::replace_data(size_t offset, size_t coun
     // 5. Insert data into node’s data after offset code units.
     // 6. Let delete offset be offset + data’s length.
     // 7. Starting from delete offset code units, remove count code units from node’s data.
-    StringBuilder builder;
-    builder.append(MUST(utf16_view.substring_view(0, offset).to_utf8(Utf16View::AllowInvalidCodeUnits::Yes)));
-    builder.append(data);
-    builder.append(MUST(utf16_view.substring_view(offset + count).to_utf8(Utf16View::AllowInvalidCodeUnits::Yes)));
-    m_data = MUST(builder.to_string());
+    Utf16Data full_data;
+    auto before_data = utf16_view.substring_view(0, offset);
+    full_data.append(before_data.data(), before_data.length_in_code_units());
+    auto inserted_data = MUST(AK::utf8_to_utf16(data));
```

Comment:
I noticed there's a `inserted_data_length_in_utf16_code_units` variable higher up in this function. Is it necessary to have that anymore? Could you replace its usages with `inserted_data.size()`?